src/fheroes2/army/army_ui_helper.cpp

```diff
@@ -44,14 +44,46 @@ void fheroes2::drawMiniMonsters( const Troops & troops, int32_t cx, const int32_
         count = troops.GetOccupiedSlotCount();
     }
 
-    const int chunk = width / count;
+    int chunk = static_cast<int32_t>( width / count );
+    size_t slots = troops.Size();
+    int slotsToSkip = 1;
+
+    int slotOffset = 1;
```

Comment:
The same comment as above :)

```suggestion
    int32_t slotsToSkip = 1;

    int32_t slotOffset = 1;
```